### PR TITLE
Add Snowflake Tracking

### PR DESCRIPTION
**Pull request by sungchun12**
See how people are using SQLMesh with Snowflake. Should look and feel like this in the query history.
image

### CODE REVIEW

File: sqlmesh/core/config/connection.py [hunk 1]
```diff
@@ -407,6 +408,7 @@ class SnowflakeConnectionConfig(ConnectionConfig):
     role: t.Optional[str] = None
     authenticator: t.Optional[str] = None
     token: t.Optional[str] = None
+    application: str = 'Tobiko_SQLMesh'
```
Comment (tobymao) [inline sqlmesh/core/config/connection.py:411]: this shouldn’t be configurable

Comment (sungchun12) [reply sqlmesh/core/config/connection.py]: made it a property

Comment (sungchun12) [reply sqlmesh/core/config/connection.py]: Nevermind, made it a classvar instead. Let me know if you want more protection on it.

File: sqlmesh/core/config/connection.py [hunk 2]
```diff
@@ -407,6 +407,7 @@ class SnowflakeConnectionConfig(ConnectionConfig):
     role: t.Optional[str] = None
     authenticator: t.Optional[str] = None
     token: t.Optional[str] = None
+    application: t.ClassVar[str] = "Tobiko_SQLMesh"
```
Comment (georgesittas) [inline sqlmesh/core/config/connection.py:410]: I think doing `t.Literal["Tobiko_SQLMesh"] = "Tobiko_SQLMesh"` is more common in the codebase, but not sure if `t.ClassVar` semantics match the use case better.

Does this guard from the field being mutated, though? Perhaps we should add it to `_FIELD_UPDATE_STRATEGY` with the `IMMUTABLE` strategy? cc @tobymao

Comment (sungchun12) [reply sqlmesh/core/config/connection.py:410]: I'll assume Iaroslav's 👍 means it's worth changing this.

**Review by tobymao — APPROVED** [review]

**Pull request merged**